compiler/modules/front/resolveTypes.bal

```diff
@@ -329,6 +350,14 @@ function resolveTypeDesc(ModuleSymbols mod, s:ModuleLevelDefn modDefn, int depth
     panic error("unimplemented type-descriptor");
 }
 
+// This defer isEmpty check if Env is not ready. But if Env is ready caller must validate type
```

Comment:
I don't see the caller checking.  In any case, this should be responsible for the complete check.

Call it something like `nonEmptyType` and return an error or t:SemType.  Then instead of 

```
            t:SemType ty = d.define(env, fields, rest);
            deferIsEmptyCheck(mod, modDefn, td, ty);
            return ty;
```

do 

```
return check nonEmptyType(mod, modDefn, td, d.define(env, fields, rest));
```

Oh you will need to pass in env too.

Reply:
fixed